File: java/src/main/java/com/experian/dvs/client/phone/validate/Metadata.java
package com.experian.dvs.client.phone.validate;

import com.experian.dvs.client.server.phone.RestApiPhoneValidateMetadata;

import java.util.Objects;

public class Metadata {

    private final String code;
    private final String message;
    private final PhoneDetail phoneDetail;

    public Metadata(final RestApiPhoneValidateMetadata metadata) {
        this.code = Objects.toString(metadata.getCode(), "");
        this.message = Objects.toString(metadata.getMessage(), "");
        this.phoneDetail = metadata.getPhoneDetail() != null ? new PhoneDetail(metadata.getPhoneDetail()) : null;

    }

    public String getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public PhoneDetail getPhoneDetail() {
        return phoneDetail;
    }
}

File: csharp/DVSClient/DVSClient/phone/validate/Metadata.cs
using DVSClient.Server.Phone;

namespace DVSClient.Phone.Validate
{
    public class Metadata
    {
        public string Code { get; }
        public string Message { get; }
        public PhoneDetail? PhoneDetail { get; }

        public Metadata(RestApiPhoneValidateMetadata metadata)
        {
            Code = metadata.Code ?? string.Empty;
            Message = metadata.Message ?? string.Empty;
            PhoneDetail = metadata.PhoneDetail != null ? new PhoneDetail(metadata.PhoneDetail) : null;
        }
    }
}
